src/components/calcite-dropdown/calcite-dropdown.tsx

```diff
@@ -134,73 +204,44 @@ export class CalciteDropdown {
   /** fires when a dropdown has been closed **/
   @Event() calciteDropdownClose: EventEmitter<void>;
 
-  @Listen("click") openDropdown(e) {
-    if (
-      this.triggers.includes(e.target) ||
-      this.triggers.some((trigger) => trigger.contains(e.target))
-    ) {
-      e.preventDefault();
-      e.stopPropagation();
-      this.openCalciteDropdown();
-    }
-  }
-
-  @Listen("click", { target: "window" }) closeCalciteDropdownOnClick(e) {
+  @Listen("click", { target: "window" })
+  closeCalciteDropdownOnClick(e: Event): void {
+    const target = e.target as HTMLElement;
     if (
       this.active &&
-      e.target.nodeName !== "CALCITE-DROPDOWN-ITEM" &&
-      e.target.nodeName !== "CALCITE-DROPDOWN-GROUP"
+      target.nodeName !== "CALCITE-DROPDOWN-ITEM" &&
+      target.nodeName !== "CALCITE-DROPDOWN-GROUP"
     ) {
       this.closeCalciteDropdown();
     }
   }
 
-  @Listen("calciteDropdownCloseRequest") closeCalciteDropdownOnEvent() {
+  @Listen("calciteDropdownCloseRequest")
+  closeCalciteDropdownOnEvent(): void {
     this.closeCalciteDropdown();
   }
 
   @Listen("calciteDropdownOpen", { target: "window" })
-  closeCalciteDropdownOnOpenEvent(e) {
+  closeCalciteDropdownOnOpenEvent(e: Event): void {
     if (e.target !== this.el) this.active = false;
   }
 
-  @Listen("keydown") keyDownHandler(e) {
-    const key = getKey(e.key);
-    if (
-      this.triggers.includes(e.target) ||
-      this.triggers.some((trigger) => trigger.contains(e.target))
-    ) {
-      if (e.target.nodeName !== "BUTTON" && e.target.nodeName !== "CALCITE-BUTTON") {
-        switch (key) {
-          case " ":
-          case "Enter":
-            this.openCalciteDropdown();
-            break;
-          case "Escape":
-            this.closeCalciteDropdown();
-            break;
-        }
-      } else if (key === "Escape" || (e.shiftKey && key === "Tab")) {
-        this.closeCalciteDropdown();
-      }
-    }
-  }
-
-  @Listen("mouseenter") mouseoverHandler() {
+  @Listen("mouseenter")
+  mouseoverHandler(): void {
     if (this.type === "hover") {
       this.openCalciteDropdown();
     }
   }
 
-  @Listen("mouseleave") mouseoffHandler() {
+  @Listen("mouseleave")
+  mouseoffHandler(): void {
```

Comment:
Nitpick: Unrelated, but the event names doesn't match their handlers. Sidebar, should the mouse event names be camelCased?